docs/references/nextjs/clerk-middleware.mdx

```diff
@@ -295,4 +295,15 @@ The `clerkMiddleware()` function accepts an optional object. The following optio
 | `signInUrl?`                       | `string`             | An alternative sign in URL.                                                                                                                                                                                                                                                                                                                     |
 | `signUpUrl?`                       | `string`             | An alternative sign up URL.                                                                                                                                                                                                                                                                                                                     |
 | `publishableKey` | `string` | The Clerk publishable key for your instance. This can be found in your Clerk Dashboard on the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys)** page.|
-| `secretKey?` | `string` | The Clerk secret key for your instance. This can be found in your Clerk Dashboard on the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys)** page.  |
+| `secretKey?` | `string` | The Clerk secret key for your instance. This can be found in your Clerk Dashboard on the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys)** page. `CLERK_ENCRYPTION_KEY` must be defined when providing `secretKey` as an option, refer to [Server-side options propagation](#server-side-options-propagation) |
+
+### Server-side options propagation
+
+The `clerkMiddleware` options listed below are propagated to the Next.js application server, making them accessible for server-side helpers like [auth](/docs/references/nextjs/auth):
```

Comment:
How to clearly clarify the constraints of redirecting URLs between client-side and server-side usage 🤔 Some options:

- Update `ClerkProvider` properties descriptions to mention that those aren't accessible on server-side helpers such as `auth`
- Update `auth` page to mention that the redirect URLs come from either environment variables or middleware options